controller-server/src/main/java/com/yahoo/vespa/hosted/controller/Application.java

```diff
@@ -0,0 +1,242 @@
+    private final Optional<IssueId> deploymentIssueId;
+    private final Optional<IssueId> ownershipIssueId;
+    private final Optional<User> owner;
+    private final OptionalInt majorVersion;
+    private final ApplicationMetrics metrics;
+    private final Optional<String> pemDeployKey;
+    private final Map<InstanceName, Instance> instances;
+
+    /** Creates an empty application. */
+    public Application(TenantAndApplicationId id, Instant now) {
+        this(id, now, DeploymentSpec.empty, ValidationOverrides.empty, Change.empty(), Change.empty(),
+             Optional.empty(), Optional.empty(), Optional.empty(), OptionalInt.empty(),
+             new ApplicationMetrics(0, 0), Optional.empty(), OptionalLong.empty(), true, List.of());
+    }
+
+    // DO NOT USE! For serialization purposes, only.
+    public Application(TenantAndApplicationId id, Instant createdAt, DeploymentSpec deploymentSpec, ValidationOverrides validationOverrides,
+                Change change, Change outstandingChange, Optional<IssueId> deploymentIssueId, Optional<IssueId> ownershipIssueId, Optional<User> owner,
+                OptionalInt majorVersion, ApplicationMetrics metrics, Optional<String> pemDeployKey, OptionalLong projectId,
+                boolean internal, Collection<Instance> instances) {
+        this.id = Objects.requireNonNull(id, "id cannot be null");
+        this.createdAt = Objects.requireNonNull(createdAt, "instant of creation cannot be null");
+        this.deploymentSpec = Objects.requireNonNull(deploymentSpec, "deploymentSpec cannot be null");
+        this.validationOverrides = Objects.requireNonNull(validationOverrides, "validationOverrides cannot be null");
+        this.change = Objects.requireNonNull(change, "change cannot be null");
+        this.outstandingChange = Objects.requireNonNull(outstandingChange, "outstandingChange cannot be null");
+        this.deploymentIssueId = Objects.requireNonNull(deploymentIssueId, "deploymentIssueId cannot be null");
+        this.ownershipIssueId = Objects.requireNonNull(ownershipIssueId, "ownershipIssueId cannot be null");
+        this.owner = Objects.requireNonNull(owner, "owner cannot be null");
+        this.majorVersion = Objects.requireNonNull(majorVersion, "majorVersion cannot be null");
+        this.metrics = Objects.requireNonNull(metrics, "metrics cannot be null");
+        this.pemDeployKey = Objects.requireNonNull(pemDeployKey, "pemDeployKey cannot be null");
+        this.projectId = Objects.requireNonNull(projectId, "projectId cannot be null");
+        this.internal = internal;
+        this.instances = instances.stream().collect(Collectors.toUnmodifiableMap(instance -> instance.id().instance(),
+                                                                                 instance -> instance));
+    }
+
+    /** Returns an aggregate application, from the given instances, if at least one. */
+    public static Optional<Application> aggregate(List<Instance> instances) {
```

Comment:
Temporary creator. Picks common data from the "default" instance, if it exists, or any instances otherwise.